### ISSUE TITLE
I got many segmentation faults, how can i write code to fix them?

### ISSUE THREAD

**Issue by yoosan** [torch/torch7#455]
It happens when i use some loop structures. In loop body there are tensor operations, I always got segmentation faults. like this

    [1]    6062 segmentation fault

**Comment by fmassa** [comment]
Can you give a minimum example reproducing the problem ?

**Comment by yoosan** [comment]
My code here 
<pre>
torch.setdefaulttensortype('torch.FloatTensor')

function split(inputstr, sep)
  if sep == nil then
    sep = "%s"
  end
  local t={} ; i=1
  for str in string.gmatch(inputstr, "([^"..sep.."]+)") do
    t[i] = str
    i = i + 1
  end
  return t
end

require 'nn'
local w2v_model = torch.load('vectors.t7')
local lookup =  w2v_model.w2vvocab
local sentence_dim = 100

size = w2v_model.M:size(2)

function sum_embedding(sentence, ii)
  if ii~=267 then return end
  print('error here')
  io.flush()
  sentence_list = split(sentence, " ")
  local sum = w2v_model.M[lookup[sentence_list[1]]]
  local s_size = #sentence_list
  for i = 2, s_size do
    if lookup[sentence_list[i]] ~= nil then
      local vecpre = w2v_model.M[lookup[sentence_list[i]]]
      sum = torch.cat({sum, vecpre})
    else
      local vecpre = torch.rand(size)
      sum = torch.cat(sum, vecpre)
    end
  end
  linear = nn.Linear(s_size * size, sentence_dim)
  sigmoid = nn.LogSigmoid()
  net = nn.Sequential()
  net:add(linear)
  net:add(sigmoid)
  res = net:forward(sum)
  return res
end

--print('here')

local input = torch.DiskFile('/Users/yoosan/Desktop/data/snli_1.0/input.txt')

local sentences_size = tonumber(input:readInt())
print(sentences_size)
local Data = torch.FloatTensor(sentences_size, sentence_dim)
print('here')

local index = 1
local pairs = {}
print('here')

local partial = math.floor(sentences_size / 1000)

print('here')

for i = 1, sentences_size do
  local s = input:readString('*l')
  local s_list = split(s, '\t')
  assert(#s_list == 3)
  pairs[index] = {s_list[1], s_list[2]}
  index = index + 1
  local s_embed = sum_embedding(s_list[1], i)
  --print(i, s)
end

print(index)



-- print(sum_embedding('today is a nice a'))

</pre>

**Comment by yoosan** [comment]
The sentences_size will be large - 542342.
I guess the fault happens when I using random float tensor, and do some operations.

**Comment by fmassa** [comment]
Can't you add a simpler example which doesn't require this `vectors.t7` file ?

**Comment by yoosan** [comment]
Hi fmassa, thank you. I solved this problem. 
The error happens just because of my stupid. The reason is that the array bounds.
